### PR TITLE
chore: check git state in lock-api-state.sh

### PR DESCRIPTION
**Pull request by stoksc**
## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by ioga — APPROVED** [review]

File: tools/scripts/lock-api-state.sh [hunk 1]
```diff
@@ -1,5 +1,11 @@
 #!/bin/bash -ex
 
+# check for dirty changes
```
Comment (dzhu) [inline tools/scripts/lock-api-state.sh:3]: There's a check already in the Makefile, I just... commented it out at some point, I guess. (Also, I think this condition 
is reversed?)

Comment (stoksc) [reply tools/scripts/lock-api-state.sh:3]: uhhh.. what makefile? I couldn't find anything calling this script?

Comment (dzhu) [reply tools/scripts/lock-api-state.sh:3]: In `proto/Makefile` under `gen-buf-image` (which this calls, rather than the other way around).

Comment (stoksc) [reply tools/scripts/lock-api-state.sh:3]: oh I see, it doesn't call the script, but the gen step has the check

Comment (stoksc) [reply tools/scripts/lock-api-state.sh:3]: I will remove from the make file and leave it here. I remember it being annoying in the make file - when you want to make a breaking change, you know you have uncommitted changes, but when you are cutting a release, you want the check.

Comment (stoksc) [reply tools/scripts/lock-api-state.sh:3]: look good to you @dzhu ?

Comment (dzhu) [reply tools/scripts/lock-api-state.sh:3]: Makes sense!

**Pull request merged**